lib/internal/crypto/keys.js

```diff
@@ -203,7 +209,7 @@ function parseKeyEncoding(enc, keyType, isPublic, objName) {
 // when this is used to parse an input encoding and must be a valid key type if
 // used to parse an output encoding.
 function parsePublicKeyEncoding(enc, keyType, objName) {
-  return parseKeyFormatAndType(enc, keyType, true, objName);
+  return parseKeyEncoding(enc, keyType, keyType ? true : undefined, objName);
```

Comment:
The code must function correctly, you tested it, but its not clear from the code near here why if `keyType === 'secret'`, the third arg, `isPublic`, is set to `true`. Maybe that can never happen? If so, an assert would make it clear.

nit: it looks like the ternary could be reexpressed as `!!keyType`.

Reply:
> The code must function correctly, you tested it, but its not clear from the code near here why if `keyType === 'secret'`, the third arg, `isPublic`, is set to `true`. Maybe that can never happen? If so, an assert would make it clear.

`keyType` refers to the type of the asymmetric key (e.g. `'rsa'` or `'dsa'`), not to `'public'`, `'private'` etc. It is mostly used to distinguish between input / output encoding and to ensure that PKCS#1 is only used if `keyType === 'rsa'` etc. Sorry about the confusion.

> nit: it looks like the ternary could be reexpressed as `!!keyType`.

I _think_ we still need all three possible values (`true`, `false`, `undefined`). `undefined` means that it can either be `true` or `false`.